pouchdb/pouchdb
# Fix pouchdb-find inconsistencies - issue #7810

## Yet another PR for issue #7810
This PR (imho) replaces #8066 and #7895.

I've took the work @pubkey on the issue, and took the liberty to expand on it. Hopefully adding value.

I spend quite some time to get to understand the _pouchdb-find_ module (or couchdb/pouchdb for that matter).
I added some more test cases for the issue, hopefully covering the desired behavior when it come to the results certain selectors should give.

All of the tests written test the behaviour on both a db with indexes, and without.
Also the testsuite now runs on the remote as well, to make sure the local and remote behave similar as well.


## How to test

Make sure you have CouchDB installed and you ran `npm install` (see [TESTING.md](https://github.com/pouchdb/pouchdb/blob/master/TESTING.md)). 
Then run:
```bash
TYPE=find PLUGINS=pouchdb-find npm run dev
```

To run all the tests in the browser visit [go here](http://127.0.0.1:8000/tests/find/?adapter=idb&plugins=pouchdb-find)

Or only the new test suite from this PR [go here](http://127.0.0.1:8000/tests/find/?adapter=idb&plugins=pouchdb-find&grep=test%5C.issue7810%5C.js)


## Feedback

Hopefully this is a stab in the right direction. But by all means feel free to share light how to improve (Code-wise, problem-solving-wise, but process-wise as well).
